dags/htan-validate-dag.py

```diff
@@ -0,0 +1,83 @@
+from datetime import datetime
+
+import synapseclient
+from airflow.decorators import dag, task
+from airflow.models import Variable
+from airflow.operators.python import get_current_context
+from services.NextflowTowerService import create_and_open_tower_workspace
+from services.AWSService import upload_file_s3
+
+
+@dag(
+    schedule_interval=None,
+    start_date=datetime(2022, 11, 11),
+    catchup=False,
+    default_args={
+        "retries": 2,
+    },
+    tags=["nextflow_tower"],
+)
+def htan_nf_dcqc_dag():
+    @task()
+    def get_synapse_input_file() -> str:
+        """
+        Gets synapse file (input csv) from synapse and saves it in buffer.
+        Passes along dict object with the path to the downloaded file and the name of the file.
+
+        Returns:
+            str: String containing the path to the downloaded file and the name of the file.
+        """
+        # simple param passing from run with config - sets my test file as default for now
+        syn_id = get_current_context()["params"].get("syn_id") or "syn50919899"
+        syn_token = Variable.get("SYNAPSE_AUTH_TOKEN")
+        syn = synapseclient.login(authToken=syn_token)
+        file_path = syn.get(syn_id).path
+        return file_path
+
+    @task()
+    def stage_input_in_s3(file_path: str) -> str:
+        """
+        Uploads file to Nextflow Tower S3 bucket, returns string path to file
+        Args:
+            file_path (str): String containing the path to the downloaded file to be uploaded to S3
+
+        Returns:
+            s3_uri (str): Path to S3 bucket location of file
+        """
+        s3_uri = upload_file_s3(file_path=file_path, bucket_name="orca-dev-project-tower-bucket")
+        return s3_uri
+
+
+    @task()
+    def launch_tower_workflow(workspace_id: str, s3_uri: str):
+        """
+        Launches tower workflow
+
+        Args:
+            tower_utils (sagetasks.nextflowtower.utils.TowerUtils): TowerUtils class instance
+            workspace_id (str): Workspace ID for tower run
+            s3_path (str): Path to S3 location of input file
+        """
+        tower_utils = create_and_open_tower_workspace(platform="sage-dev", workspace_id=workspace_id)
+        tower_utils.launch_workflow(
+            compute_env_id="635ROvIWp5w17QVdRy0jkk",
+            pipeline="Sage-Bionetworks-Workflows/nf-dcqc",
+            run_name="nf-dcqc-airflow-test",
```

Comment:
there is no problem launching multiple workflows with the same name, they both seem to launch just fine. I am going to remove the `run_name` for now though because it is not really necessary.

Reply:
And I will use underscores in the future

Reply:
If you can use the same run name for different "launches", you can set it to `airflow-{{ run_id }}` so we can more easily cross-reference Airflow and Nextflow runs.